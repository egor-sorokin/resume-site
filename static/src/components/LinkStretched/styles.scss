@import '../../styles/base/_vars.scss';

.link-stretched {
  letter-spacing: 3px;

  transform: translateX(-40px);
  transition: transform $defaultTiming $defaultEasing,
  letter-spacing $defaultTiming $defaultEasing;
}

.link-stretched:hover {
  letter-spacing: 5px;

  transform: translateX(-34px);
}
