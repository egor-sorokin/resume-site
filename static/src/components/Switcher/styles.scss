@import '../../styles/base/_vars.scss';

.switcher {
  position: absolute;
  right: 0;
  top: 0;

  width: 4vw;
}

.switcher__line {
  height: 44vh;
  width: 1px;

  background-color: $dune;
  opacity: 0.3;
}

.switcher__link {
  display: flex;

  height: 12vh;

  align-items: center;
}

.switcher--white {
  .switcher__line {
    background-color: $mercury;
    opacity: 0.6;
  }
  .link-stretched {
    color: $mercuryLight;
  }
}
