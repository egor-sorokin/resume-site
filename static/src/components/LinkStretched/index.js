import React from 'react';
import PropTypes from 'prop-types';
import './styles.scss';


const LinkStretched = props => {
  const {text, url, target, cssClasses, onClick} = props;

  const handleClick = (e) => {
    if (!url) {
      e.preventDefault();
      onClick();
    } else {
      onClick(e.currentTarget.dataset.anchorId.substring(1), true);
    }
  };

  return (
    <a
      href={url}
      data-anchor-id={url}
      target={target}
      onClick={handleClick}
      className={'link-stretched ' + cssClasses}
    >{text}</a>
  )
};


LinkStretched.propTypes = {
  onClick: PropTypes.func,
  cssClasses: PropTypes.string,
  text: PropTypes.string,
  url: PropTypes.string,
  target: PropTypes.string
};

LinkStretched.defaultProps = {
  cssClasses: '',
  text: '',
  url: '',
  target: '',
  onClick: () => {
  }
};


export default LinkStretched;