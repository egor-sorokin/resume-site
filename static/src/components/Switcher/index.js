import React from 'react';
import PropTypes from 'prop-types';
import LinkStretched from  '../LinkStretched/index'
import './styles.scss';


const Switcher = ({data, switcherLink, onClick}) => {
  let link;
  let {type, text} = switcherLink;

  if (text && text.toLowerCase() === 'about') {
    link = (
      <LinkStretched
        cssClasses="switcher__link font-s-12-secondary text-c-mercury-light"
        text={text}
        onClick={onClick}
      ></LinkStretched>
    );
  } else if ((data && data.url)) {
    link = (
      <LinkStretched
        cssClasses="switcher__link font-s-12-secondary text-c-dune"
        text={text}
        onClick={onClick}
        url={data.url}
        target="_blank"
      ></LinkStretched>
    );
  }

  return (
    <div className={`switcher switcher--${type}`}>
      <div className="switcher__line line--top"/>
      {link}
      {(link) && (<div className="switcher__line line--bottom"/>)}
    </div>);
};

Switcher.propTypes = {
  onClick: PropTypes.func,
  data: PropTypes.shape({}),
  switcherLink: PropTypes.shape({}),
};

Switcher.defaultProps = {
  onClick: () => {
  },
  data: {},
  switcherLink: {},
};

export default Switcher;
