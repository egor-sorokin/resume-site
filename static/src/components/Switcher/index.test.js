// eslint-disable-next-line
import {shallow, render, mount} from 'enzyme';
import React from 'react';
import Switcher from './index';
import LinkStretched from  '../LinkStretched/index';


describe('<Switcher />', () => {
  let component;
  const data = {
    withUrl: {
      url: 'test.url'
    },
    withoutUrl: {
      url: ''
    }
  };
  const switcherLink = {
    about: {
      text: 'about',
      type: 'white'
    },
    another: {
      text: 'test text',
      type: 'black'
    }
  };
  const onClick = jest.fn(() => {
  });


  describe('base tests', () => {
    beforeEach(() => {
      component = shallow(<Switcher />);
    });


    it('renders without crashing', () => {
      expect(component).toBeDefined();
    });


    it('matches snapshot', () => {
      expect(component).toMatchSnapshot();
    });


    it('renders 1 component', () => {
      expect(component.children()).toHaveLength(1);
    });
  });


  describe('props tests', () => {
    beforeEach(() => {
      component = mount(<Switcher />);
    });


    it('checks onClick function in props', () => {
      component.setProps({onClick: onClick});
      expect('onClick' in component.props()).toBeTruthy();
    });


    it('renders switcher link', () => {
      component.setProps({url: data.withUrl.url, switcherLink: switcherLink.another});
      expect('url' in component.props()).toBeTruthy();
      expect('switcherLink' in component.props()).toBeTruthy();
      expect(component.find(LinkStretched)).toBeTruthy();
      expect(component.find(LinkStretched).find('a').text()).toEqual(switcherLink.another.text);
      expect(component.find('div').someWhere(n => n.hasClass('line--bottom'))).toBeTruthy();
      expect(component.find('a').someWhere(n => n.hasClass('text-c-dune'))).toBeTruthy();
      expect(component.find('div').someWhere(n => n.hasClass('switcher--black'))).toBeTruthy();
    });


    it('renders only line from the top to the middle of the screen', () => {
      expect(component.find('div').contains('.line--bottom')).toBeFalsy();
      expect(component.find('div').contains(LinkStretched)).toBeFalsy();
    });


    it('checks that the current page is About', () => {
      component.setProps({switcherLink: switcherLink.about});
      expect('switcherLink' in component.props()).toBeTruthy();
      expect(component.find(LinkStretched)).toBeTruthy();
      expect(component.find(LinkStretched).find('a').text()).toEqual(switcherLink.about.text);
      expect(component.find('div').someWhere(n => n.hasClass('line--bottom'))).toBeTruthy();
      expect(component.find('a').someWhere(n => n.hasClass('text-c-mercury-light'))).toBeTruthy();
      expect(component.find('div').someWhere(n => n.hasClass('switcher--white'))).toBeTruthy();
    });
  });
});
