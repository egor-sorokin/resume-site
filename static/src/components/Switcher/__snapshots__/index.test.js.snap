// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Switcher /> base tests matches snapshot 1`] = `
<div
  className="switcher switcher--undefined"
>
  <div
    className="switcher__line line--top"
  />
</div>
`;
